src/npg_porch/endpoints/pipelines.py

```diff
@@ -84,6 +84,32 @@ async def get_pipeline(
     return pipeline
 
 
+@router.post(
+    "/{pipeline_name}/token/{token_desc}",
+    response_model=str,
+    responses={
+        status.HTTP_201_CREATED: {"description": "Token was created"},
+        status.HTTP_404_NOT_FOUND: {"description": "Pipeline not found"},
+    },
+    summary="Create a new token for a pipeline and return it.",
+    description="""
+    Returns a token for the pipeline with the given name.
+    A valid token issued for any pipeline is required for authorisation."""
+)
+async def create_pipeline_token(
+    pipeline_name: str,
+    token_desc: str,
+    db_accessor=Depends(get_DbAccessor),
+    permissions=Depends(validate)
+) -> Response:
+    try:
+        token = await db_accessor.create_pipeline_token(name=pipeline_name, desc=token_desc)
+    except NoResultFound:
+        raise HTTPException(status_code=404,
+                            detail=f"Pipeline '{pipeline_name}' not found")
+    return Response(status_code=status.HTTP_201_CREATED, content=token, media_type="text/plain")
```

Comment:
I don't care that much, but it's a bit naughty to return different content-types depending on whether the request was executed correctly. Content negotiation would be ideal, but fastapi isn't very good at this side of API compliance.